### ISSUE TITLE
Document Information Extraction in production - Without Confidence Score?

### ISSUE THREAD

**Issue by WaterKnight1998** [clovaai/donut#101]
Good afternoon,

One of the biggest drawbacks that we found is that fields doesn't have a confidence score? How is this model supposed to be used in production?

**Comment by TheSeriousProgrammer** [comment]
This issue discusses briefly regarding the same https://github.com/clovaai/donut/issues/37, however this gives out a confidence score for the whole json not for individual entities. This models predicts the whole json as an xml string (just like gpt). So I am not sure of the feasibility in extracting confidence values for specific values. One could extract the confidences of the individual tokens in value fields to get confidence scores accordingly(This should work theoretically but may take a lot of trail and error)

Searching for your doubt in existing github issues with keywords may help you 

But please be polite while asking such queries, the clovai team has done a phenomenal job and has made it opensource. if not supportive try not be harsh in your comments in the least 😊

**Comment by WaterKnight1998** [comment]
> But please be polite while asking such queries, the clovai team has done a phenomenal job and has made it opensource. if not supportive try not be harsh in your comments in the least 😊

Yes, this model is awesome, they did and incredible job. I didn't want to harsh them. 

I just wanted to know if there was an automatic way to know if the field is OK. Especially when other applications use the data extracted by the model. That way we could reduce the work of people in the loop as much as possible.

If you suggest me a better title I can update it. I have updated first message, let me know if it is ok.

**Comment by TheSeriousProgrammer** [comment]
Thanks for updating your comment, it looks fine now 🙌

As on now there is no direct method to extract confidence scores for specific fields
But each and every predicted token has a confidence score , so theoretically it should be possible to extract individual token confidence values of a given value field and average them out to get the confidence score of the value

But there is no guarantee to know if the obtained confidence score can be used to convey the correctness of a given value (I.e it may give high confidence scores for logically incorrect but syntactically correct values)

Let's wait for the author's input on the same
